File: app/src/main/java/com/elegion/android/ui/base/view/LoadingView.java
package com.elegion.android.ui.base.view;

import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;
import com.elegion.android.util.moxy.strategy.AddToEndSingleByTagStrategy;
import com.elegion.android.util.moxy.strategy.OneExecutionByTagStrategy;

/**
 * @author Mikhail Barannikov
 */
public interface LoadingView {
    @StateStrategyType(value = AddToEndSingleByTagStrategy.class, tag = "LoadingView")
    void showLoadingIndicator();

    @StateStrategyType(value = OneExecutionByTagStrategy.class, tag = "LoadingView")
    void hideLoadingIndicator();

}

File: app/src/main/kotlin/com/bankex/pay/ui/base/view/LoadingView.kt
package com.bankex.pay.ui.base.view

import com.arellomobile.mvp.viewstate.strategy.StateStrategyType
import com.bankex.pay.util.moxy.strategy.AddToEndSingleByTagStrategy
import com.bankex.pay.util.moxy.strategy.OneExecutionByTagStrategy

interface LoadingView {
    @StateStrategyType(value = AddToEndSingleByTagStrategy::class, tag = "LoadingView")
    fun showLoadingIndicator()

    @StateStrategyType(value = OneExecutionByTagStrategy::class, tag = "LoadingView")
    fun hideLoadingIndicator()
}
